pwa/src/components/applications/applicationForm.tsx

```diff
@@ -34,6 +34,7 @@ export const ApplicationForm: React.FC<ApplicationFormProps> = ({id}) => {
   const [alert, setAlert] = React.useState<Record<string, string>>(null);
   const [application, setApplication] = React.useState<IApplication>(null);
   const [showSpinner, setShowSpinner] = React.useState<boolean>(false);
+  const [saveSpinner, setSaveSpinner] = React.useState<boolean>(false);
```

Comment:
This feels a bit duplicated, consider the following two suggestions:

1. We are now using the `saveSpinner` and `showSpinner` separately from each other, it might be better to completely remove the `showSpinner` and just always use the overlay (you can maybe check this with @rubenvdlinde )
2. The name `saveSpinner` is too specific, what if we want an overlay spinner when we're deleting something, fetching something, or any other reason. Therefore, we should rename this (to e.g. `OverlaySpinner` or `LoadingOverlay`).